homeassistant/components/hue/sensor_base.py

```diff
@@ -177,19 +132,16 @@ def register_component(self, binary, async_add_entities):
             else:
                 new_sensors.append(current[api[item_id].uniqueid])
 
-        await remove_devices(
-            self.hass,
-            self.config_entry,
-            [value.uniqueid for value in api.values()],
-            current,
+        self.bridge.hass.async_create_task(
+            remove_devices(
+                self.bridge, [value.uniqueid for value in api.values()], current,
+            )
         )
 
-        async_add_sensor_entities = self._component_add_entities.get(False)
-        async_add_binary_entities = self._component_add_entities.get(True)
-        if new_sensors and async_add_sensor_entities:
-            async_add_sensor_entities(new_sensors)
-        if new_binary_sensors and async_add_binary_entities:
-            async_add_binary_entities(new_binary_sensors)
+        if new_sensors:
+            self._component_add_entities[False](new_sensors)
+        if new_binary_sensors:
+            self._component_add_entities[True](new_binary_sensors)
 
 
 class GenericHueSensor:
```

Comment:
This should inherit from `Entity` since we're using the entity api inside.

Reply:
It's a mixin.

Reply:
I don't think that matters.

Reply:
To clarify, there is no reason to use this class outside an `Entity` inheritance tree. It should just be defined as a common sensor hue parent class.

Reply:
How would multiple inheritance work then? If our mixin inherits from entity and we mix that in with binary sensor, don't we override `state` property with returning `None` as is `Entity`, while BinarySensor returns `on` or `off`.

Reply:
My bad, you are right.

<img width="419" alt="image" src="https://user-images.githubusercontent.com/1444314/73576008-4c72d580-442e-11ea-83cc-fac76e12484a.png">

I need to brush up on my multiple inheritance